### PR TITLE
Add basic executable and build system

### LINKED ISSUE #3: Add build system

### PR DESCRIPTION
**Pull request by CohenArthur**
Closes #3

### CODE REVIEW

**Review by Skallwar — APPROVED** [review]

**Comment by 08A** [comment]
@CohenArthur don't add.dep that will be removed later

**Comment by CohenArthur** [comment]
> @CohenArthur don't add dep that will be removed later

Why would `linenoise` be removed? It's a good, lightweight alternative to `readline`. It has no memory leaks and is way simpler to edit if we need

**Comment by 08A** [comment]
Readline suck but readline is installed every where.

**Comment by Skallwar** [comment]
> It's a good, lightweight alternative to readline. It has no memory leaks and is way simpler to edit if we need

Looks good to me

**Comment by CohenArthur** [comment]
But in that case we don't need to install anything since we can just compile it. It's only one, 1k LoC file. Try the example, it's got a lot of good features and is super, super lightweight. `libreadline` is around 30k (!) LoC, which is absolutely insane and, imo, against the philosophy of this shell

**Comment by Skallwar** [comment]
If it's statically link there is no problem of installed or not

File: include/color.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * @file color.h Defines helper macros around color gestion
+ *
+ * The colors should stay within the 8-16 bit color range for compatibility with most
+ * terminals, at least for the base version of `ubsh`.
+ */
+
+#define RESET       "\x1B[0m"
+
+#define GREEN(__S)       "\x1B[32m"__S RESET
```
Comment (n1tram1) [inline include/color.h:13]: I like this

File: src/prompt.c [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#include "color.h"
+#include "prompt.h"
+
+char *prompt_get(void) {
```
Comment (n1tram1) [inline src/prompt.c:4]: `const char *` ?

File: include/prompt.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * @file prompt.h Prompt generating library
+ */
+
+/**
+ * Generate a prompt and return it
+ *
+ * @return The generated prompt
+ */
+char *prompt_get(void);
```
Comment (n1tram1) [inline include/prompt.h:13]: `const char *`

**Review by n1tram1 — CHANGES_REQUESTED** [review]

**Review by Skallwar — APPROVED** [review]

**Review by n1tram1 — APPROVED** [review]

**Pull request merged**